images/vm-route-forge/cmd/vm-route-forge/app/root.go

```diff
@@ -0,0 +1,200 @@
+	// be propagated through the whole operator, generating
+	// uniform and structured logs.
+	logf.SetLogger(zap.New(zap.Level(zapcore.Level(-1*verbosity)), zap.UseDevMode(debug)))
+}
+
+func run(opts options.Options) error {
+	setupLogger(opts.Verbosity)
+	var parsedCIDRs []*net.IPNet
+	for _, cidr := range opts.Cidrs {
+		_, parsedCIDR, err := net.ParseCIDR(cidr)
+		if err != nil || parsedCIDR == nil {
+			log.Error(err, "failed to parse passed CIDRs")
+			return err
+		}
+		parsedCIDRs = append(parsedCIDRs, parsedCIDR)
+	}
+	log.Info(fmt.Sprintf("Got CIDRs to manage: %+v", opts.Cidrs))
+
+	if opts.DryRun {
+		log.Info("Dry run mode is enabled, will not change network rules and routes")
+	}
+
+	tableID := netlinkmanager.DefaultCiliumRouteTable
+	tableIDStr := opts.TableID
+	if tableIDStr != "" {
+		tableId, err := strconv.ParseInt(tableIDStr, 10, 32)
+		if err != nil {
+			log.Error(err, "failed to parse Cilium table id, should be integer")
+			return err
+		}
+		tableID = int(tableId)
+	}
+	log.Info(fmt.Sprintf("Use cilium route table id %d", tableID))
+
+	// Load configuration to connect to Kubernetes API Server.
+	kubeCfg, err := config.GetConfig()
+	if err != nil {
+		log.Error(err, "Failed to load Kubernetes config")
+		return err
+	}
+	kubeClient, err := kubernetes.NewForConfig(kubeCfg)
+	if err != nil {
+		log.Error(err, "Failed to het kubeclient")
+		return err
+	}
+
+	ctx := signals.SetupSignalHandler()
+
+	vmSharedInformerFactory, err := informer.VirtualizationInformerFactory(kubeCfg)
+	if err != nil {
+		log.Error(err, "Failed to create informer factory")
+		return err
+	}
+	go vmSharedInformerFactory.Virtualization().V1alpha2().VirtualMachines().Informer().Run(ctx.Done())
+
+	kubeSharedInformerFactory, err := informer.KubernetesInformerFactory(kubeCfg)
+	if err != nil {
+		log.Error(err, "Failed to create node factory")
+		return err
+	}
+	go kubeSharedInformerFactory.Core().V1().Nodes().Informer().Run(ctx.Done())
+
+	ciliumSharedInformerFactory, err := informer.CiliumInformerFactory(kubeCfg)
+	if err != nil {
+		log.Error(err, "Failed to create cilium factory")
+		return err
+	}
+	go ciliumSharedInformerFactory.Cilium().V2().CiliumNodes().Informer().Run(ctx.Done())
+
+	sharedCache := cache.NewCache()
+
+	netMgr := netlinkmanager.New(vmSharedInformerFactory.Virtualization().V1alpha2().VirtualMachines(),
+		ciliumSharedInformerFactory.Cilium().V2().CiliumNodes(),
+		sharedCache,
+		log,
+		tableID,
+		parsedCIDRs,
+		opts.DryRun,
+	)
+
+	err = preRunSync(ctx, netMgr)
+	if err != nil {
+		log.Error(err, "Failed to run pre sync")
+		return err
+	}
+	routeCtrl, err := route.NewRouteController(ctx,
+		vmSharedInformerFactory.Virtualization().V1alpha2().VirtualMachines(),
+		kubeSharedInformerFactory.Core().V1().Nodes(),
+		netMgr,
+		sharedCache,
+		parsedCIDRs,
+		&log,
+	)
+	if err != nil {
+		log.Error(err, "Failed to create route controller")
+		return err
+	}
+	go routeCtrl.Run(ctx, 1)
```

Comment:
What is magic "1"? Add const to describe what is it.

Reply:
fixed